programs/govern/src/proposal.rs

```diff
@@ -34,98 +31,40 @@ pub enum ProposalState {
-    /// Vote to make the proposal pass.
-    For = 2,
-    /// This vote does not count as a `For` or `Against`, but it still contributes to quorum.
-    Abstain = 3,
-}
-
-impl Default for VoteSide {
-    fn default() -> Self {
-        VoteSide::Pending
-    }
-}
-
-impl From<VoteSide> for u8 {
-    fn from(side: VoteSide) -> Self {
-        side as u8
-    }
-}
-
-impl TryFrom<u8> for VoteSide {
-    type Error = Error;
-
-    fn try_from(value: u8) -> Result<Self> {
-        match value {
-            0 => Ok(VoteSide::Pending),
-            1 => Ok(VoteSide::Against),
-            2 => Ok(VoteSide::For),
-            3 => Ok(VoteSide::Abstain),
-            _ => program_err!(InvalidVoteSide),
-        }
-    }
-}
-
 impl Default for ProposalState {
     fn default() -> Self {
         Self::Draft
     }
 }
 
 impl Proposal {
-    /// Subtracts from the total weight of a vote for a [Proposal].
-    pub(crate) fn subtract_vote_weight(
-        &mut self,
-        vote_side: VoteSide,
-        vote_weight: u64,
-    ) -> Result<()> {
-        if vote_weight == 0 {
+    // /// Subtracts from the total weight of a vote for a [Proposal].
+    pub(crate) fn subtract_vote_weight(&mut self, side: u8, voting_power: u64) -> Result<()> {
+        if voting_power == 0 {
             return Ok(());
         }
-        match vote_side {
-            VoteSide::Pending => {}
-            VoteSide::Against => {
-                self.against_votes = unwrap_int!(self.against_votes.checked_sub(vote_weight));
-            }
-            VoteSide::For => {
-                self.for_votes = unwrap_int!(self.for_votes.checked_sub(vote_weight));
-            }
-            VoteSide::Abstain => {
-                self.abstain_votes = unwrap_int!(self.abstain_votes.checked_sub(vote_weight));
-            }
+        if side > self.max_option {
+            return Err(InvalidVoteSide.into());
         }
+        let current_vote: u64 = self.option_votes[side as usize];
+        self.option_votes[side as usize] = unwrap_int!(current_vote.checked_sub(voting_power));
         Ok(())
     }
 
     /// Adds to the total weight of a vote for a [Proposal].
-    pub(crate) fn add_vote_weight(&mut self, vote_side: VoteSide, vote_weight: u64) -> Result<()> {
-        if vote_weight == 0 {
+    pub(crate) fn add_vote_weight(&mut self, side: u8, voting_power: u64) -> Result<()> {
+        if voting_power == 0 {
             return Ok(());
         }
-        match vote_side {
-            VoteSide::Pending => {}
-            VoteSide::Against => {
-                self.against_votes = unwrap_int!(self.against_votes.checked_add(vote_weight));
-            }
-            VoteSide::For => {
-                self.for_votes = unwrap_int!(self.for_votes.checked_add(vote_weight));
-            }
-            VoteSide::Abstain => {
-                self.abstain_votes = unwrap_int!(self.abstain_votes.checked_add(vote_weight));
-            }
+        if side > self.max_option {
```

Comment:
Should be `>=` ?
1. An OptionProposal created with 2 max options, which represent ABSENT, YES, NO. OptionProposal stored max_option = 2
2. Vote for NO, the index of NO is 2, it failed with InvalidVoteSide.

Reply:
Abstain is 0, No is 2, then if they input side = 2, it will pass?

Reply:
Sorry my bad.